### DISCUSSION TITLE
Is there any options to upload ISO file via cockpit machines GUI?

### DISCUSSION THREAD

**Discussion by MRajibH** [cockpit-project/cockpit-machines#1637] [Q&A] [answered]
I was looking for an GUI based ISO file upload in cockpit but not find any.

**Comment by garrett** [comment, marked as answer]
We're currently working on a file manager for Cockpit, calling it "[Cockpit Files](https://github.com/cockpit-project/cockpit-files/)", which is a successor to "[Cockpit Navigator](https://github.com/45Drives/cockpit-navigator)", which was a third party file manger project that hasn't been updated in a while. Cockpit Files hasn't been officially released yet, as it's still a work in progress... but it should probably be released `soon`. We're actively working on the minimum feature set and polish it needs to have an official release and hope to have something to announce in the very near future.

We have an in-progress pull request that's actively being worked on for uploading files @ https://github.com/cockpit-project/cockpit-files/pull/384 — so this is actually something we're actively working on right now.

It's not specifically for Cockpit Machines, as it is a more generic way to upload files, but it would definitely work for uploading an ISO file to the remote system. Having some integration between the two would be nice, but it'd come later, if so.

(For reference, our tests upload an iso file to `/var/lib/libvirt/images/` — that's probably the ideal place to upload it, but I think it should work from anywhere Cockpit Machines would have access to via your account.)

Meanwhile, you can upload files using SSH with either SCP or SFTP. In addition to typical command line usage, many graphical file managers support file transfers over SSH. GNOME's file manager (Nautilus) and KDE's file manager (Dolphin) both natively support browsing and copying files to/from remote hosts. There's also a very detailed, cross platform one called FileZilla that's available for Linux, Windows, and macOS. For Windows, WinSCP is a popular free software choice. As far as I know, everything on macOS is proprietary and costs money except for FileZilla (or using `scp` at the command line, similar to the command line on Linux).

- GNOME Files (Nautilus) https://help.gnome.org/users/gnome-help/stable/nautilus-connect.html.en (basically: `ssh://username@servername.example.com/folder` in the location entry — the username is optional if it's your same username; the foldername is also optional... and it supports both `ssh` and `sftp` as protocols in the URI.)
- KDE file manager (Dolphin) https://userbase.kde.org/File_transfers (it's basically similar to Nautilus, where you type a URI in the path bar; except it seems you'd want to use `sftp` instead of `ssh`)
- WinSCP https://winscp.net/eng/index.php
- FileZilla https://filezilla-project.org/download.php?type=client

I'd strongly suggest using your file manager, if it supports it.

**Reply by mac2net** [reply]
I am very interested in this. In fact I was going to suggest this add-on myself. 
I am going to install it on a spare Fedora 40 server VM and Rawhide.

**Comment by mac2net** [comment]
You can also install some distros with `Download an OS` and also by providing a URL inside the Cockpit GUI.
<img width="826" alt="image" src="https://github.com/cockpit-project/cockpit-machines/assets/7514835/94a6477c-890a-49d0-8870-c68faef1a889">
<img width="823" alt="image" src="https://github.com/cockpit-project/cockpit-machines/assets/7514835/ce3159c5-d1e3-48d3-9c38-e7374e6cfa14">

**Reply by mac2net** [reply]
One more option is to navigate to `/var/lib/libvirt/boot/ ` in either Cockpit's terminal or via SSH and use `wget` to download the ISO.

**Comment by luckied** [comment]
hello, i see this as merged but on rhel 8 i don't have this option anywhere, do i need to upgrade to rhel9?
